test/new-e2e/tests/agent-shared-components/hostname/imdsv2_transition_common_test.go

```diff
@@ -0,0 +1,195 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package hostname contains helpers and e2e tests of the flare command
+package hostname
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"encoding/json"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/DataDog/test-infra-definitions/scenarios/aws/ec2"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+	awshost "github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments/aws/host"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client/agentclient"
+)
+
+type Meta struct {
+	Hostname                 string `json:"hostname"`
+	LegacyResolutionHostname string `json:"legacy-resolution-hostname,omitempty"`
+}
+
+type baseHostnameSuite struct {
+	e2e.BaseSuite[environments.Host]
+
+	hostnameMetadata Meta
+}
+
+func TestBaseHostnameSuite(t *testing.T) {
+	e2e.Run(t, &baseHostnameSuite{}, e2e.WithProvisioner(awshost.ProvisionerNoFakeIntake()))
+}
+
+func requestAgentHostnameMetadataPayload(v *baseHostnameSuite) {
+	v.T().Helper()
+
+	// Get the hostname and legacy-resolution-hostname from the agent status
+	statusArgs := agentclient.WithArgs([]string{"header", "--json"})
+	statusPayload := v.Env().Agent.Client.Status(statusArgs)
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** Did you thought of using `datadog-agent hostname` command ? It avoids the need to‏ parse the status output.
```suggestion
	// Get the hostname and legacy-resolution-hostname from the agent status
	statusArgs := agentclient.WithArgs([]string{"header", "--json"})
	hostname := v.Env().Agent.Client.Hostname()
```

Reply:
I do not think that we can retrieve the legacy_resolution_hostname with the hostname command as it output only the hostname used by the agent